### PR TITLE
CO-1507 fix review comments

### PR DESCRIPTION
**Pull request by Ruefa**

### CODE REVIEW

File: api/v1/db/oracledb/reviews-dao.js [hunk 1]
```diff
@@ -4,18 +4,24 @@ const _ = require('lodash');
...
+  // gameIds is special since the values are parsed directly into the query string
+  const paramsToFilter = ['page[size]', 'page[number]', 'gameIds'];
+  const acceptedParams = openapi.paths['/reviews'].get.parameters.map(x => x.name).filter(param => !paramsToFilter.includes(param));
 
-  // get parameters from query
-  _.forEach(Object.keys(queries), (param) => {
-    if (param !== 'gameIds' && param !== 'page[size]' && param !== 'page[number]') {
+  // pick parameters specified in openapi (getReviewsParameters) from passed in queries list
+  const sqlParams = {};
+  _.forEach(acceptedParams, (param) => {
```
Comment (wilsonia) [inline api/v1/db/oracledb/reviews-dao.js]: This for loop can be replaced by 
`const sqlParams = _.pick(queries, acceptedParams);`

**Review by wilsonia — CHANGES_REQUESTED** [review]

**Comment by wilsonia** [comment]
The following requests should have returned 400, but returned 500:
https://localhost:8180/api/v1/reviews?gameIds=[1,2]
https://localhost:8180/api/v1/reviews?gameIds=notanarray

This request returns two 400 errors, but neither accurately describe why the request is bad:
https://localhost:8180/api/v1/reviews?scoreMax=notanumber

**Comment by wilsonia** [comment]
Could you add example queries in the openapi doc for `reviewDate` and `gameIds`? Both took guessing at the required format. Their 400 responses could also indicate the required format.

File: api/v1/db/oracledb/reviews-dao.js (hunk 1 above)
Comment (Ruefa) [reply api/v1/db/oracledb/reviews-dao.js]: I spent way to much time messing with pick and omit for it to be that easy.

**Comment by Ruefa** [comment]
@wilsonia I am having trouble resolving the issue with the scoreMax parameter 

> This request returns two 400 errors, but neither accurately describe why the request is bad:
> https://localhost:8180/api/v1/reviews?scoreMax=notanumber

I tried adding a pattern to the parameter but it was ignored (I think those only work with strings) and the error message that is returned happens before any of my code is hit so I can't validate it in javascript. Do you have any ideas?

**Comment by wilsonia** [comment]
> I tried adding a pattern to the parameter but it was ignored (I think those only work with strings) and the error message that is returned happens before any of my code is hit so I can't validate it in javascript. Do you have any ideas?

It looks like that error is coming from the `runtime-error` module of the skeleton: https://github.com/osu-mist/game-curator-api/blob/d8fb9d8263b0954a8db7c1cd411bf577b2ba05b0/middlewares/runtime-errors.js#L55

This seems like a bug in the skeleton: type is not checked here, and it should be checked before all other errors, including the general one that is currently returned.

**Review by wilsonia — APPROVED** [review]

**Pull request merged**